src/sensor/pingsensor.h

```diff
@@ -0,0 +1,210 @@
+#pragma once
+
+#include "sensor.h"
+
+/**
+ * @brief Abstract ping sensors
+ *
+ */
+class PingSensor : public Sensor
+{
+    Q_OBJECT
+public:
+    /**
+     * @brief Construct a new Ping Sensor object
+     *
+     */
+    PingSensor();
+
+    /**
+     * @brief Destroy the Ping Sensor object
+     *
+     */
+    ~PingSensor();
+
+    /**
+     * @brief debug function
+     */
+    virtual void printStatus() const;
+
+    /**
+     * @brief Add new connection
+     *
+     * @param connType connection type
+     * @param connString arguments for the new connection
+     */
+    Q_INVOKABLE virtual void connectLink(AbstractLinkNamespace::LinkType connType, const QStringList& connString) = 0;
+
+    /**
+     * @brief Get device source ID
+     *
+     * @return uint8_t source id
+     */
+    uint8_t srcId() const { return _srcId; }
+    Q_PROPERTY(int srcId READ srcId NOTIFY srcIdUpdate)
+
+    /**
+     * @brief Return destiny ID
+     *
+     * @return uint8_t destiny id
+     */
+    uint8_t dstId() const { return _dstId; }
+    Q_PROPERTY(int dstId READ dstId NOTIFY dstIdUpdate)
+
+    //TODO: move functions from snake case to camel case
+    /**
+     * @brief Return firmware major version
+     *
+     * @return uint8_t firmware major version number
+     */
+    uint8_t firmware_version_major() const { return _firmware_version_major; }
+    Q_PROPERTY(int firmware_version_major READ firmware_version_major NOTIFY firmwareVersionMajorUpdate)
+
+    /**
+     * @brief Return firmware minor version
+     *
+     * @return uint8_t firmware minor version number
+     */
+    uint16_t firmware_version_minor() const { return _firmware_version_minor; }
+    Q_PROPERTY(int firmware_version_minor READ firmware_version_minor NOTIFY firmwareVersionMinorUpdate)
+
+    /**
+     * @brief Return firmware patch version
+     *
+     * @return uint8_t firmware patch version number
+     */
+    uint16_t firmware_version_patch() const { return _firmware_version_patch; }
+    Q_PROPERTY(int firmware_version_patch READ firmware_version_patch NOTIFY firmwareVersionPatchUpdate)
+
+    /**
+     * @brief Return device type number
+     *
+     * @return uint8_t Device type number
+     */
+    uint8_t device_type() const { return _device_type; }
+    Q_PROPERTY(int device_type READ device_type NOTIFY deviceTypeUpdate)
+
+    /**
+     * @brief Return device model number
+     *
+     * @return uint8_t Device model number
+     */
+    uint8_t device_model() const { return _device_model; }
```

Comment:
The [comment](https://github.com/bluerobotics/ping-viewer/pull/516/files#r292068257) was intended to go here:

> device_revision: https://github.com/bluerobotics/ping-protocol#4-device_information